specification/loadtestservice/data-plane/Microsoft.LoadTestService/preview/2023-04-01-preview/loadtestservice.json

```diff
@@ -0,0 +1,3301 @@
+        "action": {
+          "description": "Action taken after the threshold is met. Default is ‘continue’.",
+          "default": "continue",
+          "enum": [
+            "continue",
+            "stop"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "PFAction",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "continue",
+                "description": "Test will continue to run even if pass fail metric criteria metric gets failed"
+              },
+              {
+                "value": "stop",
+                "description": "Test run will stop if pass fail criteria metric is not passed."
+              }
+            ]
+          }
+        },
+        "actualValue": {
+          "format": "double",
+          "description": "The actual value of the client metric for the test run.",
+          "type": "number",
+          "readOnly": true
+        },
+        "result": {
+          "description": "Outcome of the test run.",
+          "enum": [
+            "passed",
+            "undetermined",
+            "failed"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "PFResult",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "passed",
+                "description": "Given pass fail criteria metric has passed."
+              },
+              {
+                "value": "undetermined",
+                "description": "Given pass fail criteria metric couldn't determine."
+              },
+              {
+                "value": "failed",
+                "description": "Given pass fail criteria metric has failed."
+              }
+            ]
+          }
+        }
+      }
+    },
+    "ResourceMetric": {
+      "description": "Associated metric definition for particular metrics of the azure resource ( Refer : https://docs.microsoft.com/en-us/rest/api/monitor/metric-definitions/list#metricdefinition).",
+      "required": [
+        "aggregation",
+        "metricNamespace",
+        "name",
+        "resourceId",
+        "resourceType"
+      ],
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Unique name for metric.",
+          "type": "string",
+          "readOnly": true
+        },
+        "resourceId": {
+          "description": "Azure resource id.",
+          "minLength": 1,
+          "type": "string"
+        },
+        "metricNamespace": {
+          "description": "Metric name space.",
+          "minLength": 1,
+          "type": "string"
+        },
+        "displayDescription": {
+          "description": "Metric description.",
+          "type": "string"
+        },
+        "name": {
+          "description": "The invariant value of metric name",
+          "minLength": 1,
+          "type": "string"
+        },
+        "aggregation": {
```

Comment:
We have received some customer feedback, where customer needs to query over multiple aggregation. Currently this API support for single aggregation as string. How can we extend this now? Let say if we add a new field which takes list of aggregation then it might not be a breaking change but the old field will become redundant. Wanted to your suggestion.

Reply:
Are there fixed values for "aggregation", like "min", "max", "sum", etc?  If so and if none of these has a comma character, you could -- compatibly -- allow this field to contain a comma-separated list of aggregations.